From: George Brooke <figgyc@figgyc.uk>
Subject: [PATCH] mtd: spi-nor: add support for Boya BY25Q128AS

Adds support for Boya Microelectronics (BoyaMicro) BY25Q128AS 128 MiB flash.
Tested on Creality WB-01 embedded device which uses this chip.

Signed-off-by: George Brooke <figgyc@figgyc.uk>
---
 drivers/mtd/spi-nor/spi-nor.c | 4 ++++
 1 file changed, 6 insertions(+)

--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2171,6 +2171,12 @@ static const struct flash_info spi_nor_i
 
 	{ "at45db081d", INFO(0x1f2500, 0, 64 * 1024, 16, SECT_4K) },
 
+	/* Boya */
+	{ "by25q128as", INFO(0x684018, 0, 64 * 1024, 256,
+			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB)
+	},
+
 	/* EON -- en25xxx */
 	{ "en25f32",    INFO(0x1c3116, 0, 64 * 1024,   64, SECT_4K) },
 	{ "en25p32",    INFO(0x1c2016, 0, 64 * 1024,   64, 0) },
